recorder/src/main/cpp/mapping_parser.hh

```diff
@@ -0,0 +1,80 @@
+#ifndef MAPPING_PARSER_H
+#define MAPPING_PARSER_H
+
+#include <boost/config/warning_disable.hpp>
+#include <boost/spirit/include/qi.hpp>
+#include <boost/spirit/include/phoenix_core.hpp>
+#include <boost/spirit/include/phoenix_operator.hpp>
+#include <boost/spirit/include/phoenix_object.hpp>
+#include <boost/fusion/include/adapt_struct.hpp>
+#include <boost/fusion/include/io.hpp>
+#include <boost/variant.hpp>
+
+namespace MRegion {
+    struct Range {
+        std::string start;
+        std::string end;
+    };
+
+    struct Dev {
+        std::uint32_t maj;
```

Comment:
Yes, its only for parsing purpose as of now (we may need to use it later).

maj is not a typo, if that is what you meant. It is short for major (devices are identified with major and minor number in Linux). Similarly min is minor (not minimum).